test/integrations/devicecontext.test.ts

```diff
@@ -0,0 +1,318 @@
+    });
+  });
+
+  it('add native level', async () => {
+    (
+      await executeIntegrationWith({
+        nativeContexts: { level: <SeverityLevel>'fatal' },
+      })
+    ).expectEvent.toStrictEqualToNativeContexts();
+  });
+
+  it('do not overwrite event level', async () => {
+    (
+      await executeIntegrationWith({
+        nativeContexts: { level: 'native-level' },
+        mockEvent: { level: 'info' },
+      })
+    ).expectEvent.toStrictEqualMockEvent();
+  });
+
+  it('add native environment', async () => {
+    (
+      await executeIntegrationWith({
+        nativeContexts: { environment: 'native-environment' },
+      })
+    ).expectEvent.toStrictEqualToNativeContexts();
+  });
+
+  it('do not overwrite event environment', async () => {
+    (
+      await executeIntegrationWith({
+        nativeContexts: { environment: 'native-environment' },
+        mockEvent: { environment: 'event-environment' },
+      })
+    ).expectEvent.toStrictEqualMockEvent();
+  });
+
+  it('use only native breadcrumbs', async () => {
+    const { processedEvent } = await executeIntegrationWith({
+      nativeContexts: { breadcrumbs: [{ message: 'duplicate-breadcrumb' }, { message: 'native-breadcrumb' }] },
+      mockEvent: { breadcrumbs: [{ message: 'duplicate-breadcrumb' }, { message: 'event-breadcrumb' }] },
+    });
+    expect(processedEvent).toStrictEqual({
+      breadcrumbs: [{ message: 'duplicate-breadcrumb' }, { message: 'native-breadcrumb' }]
+    });
+  });
+
+  async function executeIntegrationWith({
+    nativeContexts,
+    mockEvent,
+  }: {
+    nativeContexts: Record<string, unknown>;
+    mockEvent?: Event;
+  }): Promise<{
+    processedEvent: Event | null;
+    expectEvent: {
+      toStrictEqualToNativeContexts: () => void;
+      toStrictEqualMockEvent: () => void;
+    };
+  }> {
+    (
+      NATIVE.fetchNativeDeviceContexts as jest.MockedFunction<typeof NATIVE.fetchNativeDeviceContexts>
+    ).mockImplementation(() => Promise.resolve(nativeContexts as NativeDeviceContextsResponse));
+    const originalNativeContexts = { ...nativeContexts };
+    const originalMockEvent = { ...mockEvent };
+    const processedEvent = await executeIntegrationFor(mockEvent ?? {});
+    return {
+      processedEvent,
+      expectEvent: {
+        toStrictEqualToNativeContexts: () => expect(processedEvent).toStrictEqual(originalNativeContexts),
+        toStrictEqualMockEvent: () => expect(processedEvent).toStrictEqual(originalMockEvent),
+      },
+    };
+  }
+
+  function executeIntegrationFor(mockedEvent: Event): Promise<Event | null> {
+    return new Promise((resolve, reject) => {
+      integration.setupOnce(async eventProcessor => {
+        try {
+          const processedEvent = await eventProcessor(mockedEvent, {});
+          resolve(processedEvent);
+        } catch (e) {
+          reject(e);
+        }
+      }, mockGetCurrentHub);
+    });
+  }
+});
+
+describe('Device Context Breadcrumb filter', () => {
```

Comment:
I think test like this would not pass, but from the code I understand that the breadcrumbs should always be ordered the same in both arrays?

```js
const jsList = [
  { timestamp: 2, message: 'new js' },
  { timestamp: 1, message: 'new js' },
  { timestamp: 3, message: 'new js' },
  { timestamp: 4, message: 'new js' },
] as Breadcrumb[];
const nativeList = [
  { timestamp: 2, message: 'new native' },
  { timestamp: 1, message: 'new js' },
  { timestamp: 2, message: 'new js' },
  { timestamp: 3, message: 'new native' },
  { timestamp: 4, message: 'new native' },
] as Breadcrumb[];
```